### ISSUE TITLE
System.ArgumentOutOfRangeException: The value must be greater than or equal to zero and less than the console's buffer size in that dimension.

### ISSUE THREAD

**Issue by parseen254** [PowerShell/PSReadLine#2377]
Last 18 Keys:
 n p m Space i Enter
 Ctrl+l Ctrl+l n o Backspace p m Space i Enter
 Enter
 Ctrl+l

Exception:
System.ArgumentOutOfRangeException: The value must be greater than or equal to zero and less than the console's buffer size in that dimension.
Parameter name: top
Actual value was -3.
   at System.Console.SetCursorPosition(Int32 left, Int32 top)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(ConsoleKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics)

**Comment by iSazonov** [comment]
Windows Terminal?

**Comment by PieterHartzer** [comment]
Opened PowerShell from Windows Explorer in a local drive shared through remote desktop. The path is rather long.

```Oops, something went wrong.  Please report this bug with the details below.
Report on GitHub: https://github.com/lzybkr/PSReadLine/issues/new
-----------------------------------------------------------------------
Last 2 Keys:
 d o

Exception:
System.ArgumentOutOfRangeException: The value must be greater than or equal to zero and less than the console's buffer size in that dimension.
Parameter name: left
Actual value was -1.
   at System.Console.SetCursorPosition(Int32 left, Int32 top)
   at Microsoft.PowerShell.Internal.VirtualTerminal.set_CursorLeft(Int32 value)
   at Microsoft.PowerShell.PSConsoleReadLine.ReallyRender(RenderData renderData, String defaultColor)
   at Microsoft.PowerShell.PSConsoleReadLine.ForceRender()
   at Microsoft.PowerShell.PSConsoleReadLine.Insert(Char c)
   at Microsoft.PowerShell.PSConsoleReadLine.SelfInsert(Nullable`1 key, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(ConsoleKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics)
-----------------------------------------------------------------------
PS Microsoft.PowerShell.Core\FileSystem::\\tsclient\C\src\<REMOVED>\<REMOVED>\bin\Release\netcoreapp2.1> t```

**Comment by m-blattner** [comment]
For me I opened two PowerShell Windows. Then moved them both using Windows+ArrowKeys to my third screen. Next I presses UpArrow inside one of the PS Windows to get the last command and that is when I got the error.

(Sry but output is in german)

```ps
Bericht auf GitHub: https://github.com/lzybkr/PSReadLine/issues/new
-----------------------------------------------------------------------
Letzte 2 Tasten:
 ` UpArrow

Ausnahme:
System.ArgumentOutOfRangeException: Der Wert muss größer als oder gleich 0 (null) sein und geringer als die Puffergröße der Konsole in dieser Dimension.
Parametername: top
Der tatsächliche Wert war -1.
   bei System.Console.SetCursorPosition(Int32 left, Int32 top)
   bei Microsoft.PowerShell.PSConsoleReadLine.ReallyRender(RenderData renderData, String defaultColor)
   bei Microsoft.PowerShell.PSConsoleReadLine.ForceRender()
   bei Microsoft.PowerShell.PSConsoleReadLine.HistoryRecall(Int32 direction)
   bei Microsoft.PowerShell.PSConsoleReadLine.PreviousHistory(Nullable`1 key, Object arg)
   bei Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(ConsoleKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   bei Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   bei Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics)
```

Edit: I noticed there is already a pinned Issue referencing these problems, will look into that.
Edit: Fine now after updating PSReadLine 2.2.0-beta2!

**Comment by ngetchell** [comment]
Similar issue here. 

``` powershell
Oops, something went wrong.  Please report this bug with the details below.
Report on GitHub: https://github.com/lzybkr/PSReadLine/issues/new
-----------------------------------------------------------------------
Last 11 Keys:
 d i s a b l e - = Backspace Ctrl+Space

Exception:
System.ArgumentOutOfRangeException: The value must be greater than or equal to zero and less than the console's buffer size in that dimension.
Parameter name: top
Actual value was -15.
   at System.Console.SetCursorPosition(Int32 left, Int32 top)
   at Microsoft.PowerShell.PSConsoleReadLine.Menu.EnsureMenuAndInputIsVisible(IConsole console, Int32 tooltipLineCount)
   at Microsoft.PowerShell.PSConsoleReadLine.Menu.DrawMenu(Menu previousMenu)
   at Microsoft.PowerShell.PSConsoleReadLine.PossibleCompletionsImpl(CommandCompletion completions, Boolean menuSelect)
   at Microsoft.PowerShell.PSConsoleReadLine.CompleteImpl(Boolean menuSelect)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(ConsoleKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics)
-----------------------------------------------------------------------
```

Version Information:

``` powershell
PS C:\Users\ngetchell> Get-Module psreadline

ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     2.0.0      PSReadline                          {Get-PSReadLineKeyHandler, Get-PSReadLineOption, Remove-PS...


PS C:\Users\ngetchell> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.19041.906
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.906
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by daxian-dbw** [comment]
The `Cltr+l` issue was fixed (see #1315), and the other `ArgumentOutOfRangeException` issues reported here were duplicates of #1306, which was also fixed.

Please upgrade to the 2.1.0 version of PSReadLine from PowerShellGallery. See the [upgrading section](https://github.com/PowerShell/PSReadLine#upgrading) for instructions. Please let us know if you run into the same issue with the latest version.
